.NET/Microsoft.Recognizers.Text.DateTime/Utilities/TimeFunctions.cs

```diff
@@ -167,11 +168,19 @@ public void AddDesc(TimeResult result, string dayDesc)
 
             dayDesc = NormalizeDayDesc(dayDesc);
 
-            if (LowBoundDesc.ContainsKey(dayDesc) && result.Hour < LowBoundDesc[dayDesc])
+            if (result.Hour >= 0 && LowBoundDesc.ContainsKey(dayDesc) && (result.Hour < LowBoundDesc[dayDesc] ||

```

Comment:
Can you add an overall comment here to clarify what this piece of code handles? Cases with time-of-day (like afternoon), right?

Reply:
ok, will do

Reply:
done